qiskit/utils/quantum_instance.py

```diff
@@ -274,7 +274,12 @@ def __init__(
                 stacklevel=2,
             )
 
-        run_config = RunConfig(shots=shots, max_credits=max_credits)
+        # Remove this warnings filter when removing the deprecated `max_credits` arg. We
+        # filter here so that we don't "double warn" if max_credits was set.
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", DeprecationWarning)
+            run_config = RunConfig(shots=shots, max_credits=max_credits)
```

Comment:
A bunch of tests were failing because we unconditionally set the `max_credits` arg. Before, we would only warn if `max_credits is not None`, meaning that `max_credits=None` would be fine, even though it will soon be an error to set `max_credits` to anything.

Additionally, if you set `max_credits` here, you'd get a warning both from `QuantumInstance.__init__` and `RunConfig.__init__`, even though the user is not directly creating `RunConfig`.

So, I think the better fix is to filter out the warning from `RunConfig` and only use `QuantumInstance`'s.

Reply:
this is definitely true, though I'm not too concerned about this unless `None` wasn't already the default.  _Technically_ it'll have been possible to use a calling convention that will be broken in a later Terra without a warning, but if you were never passing _anything_ other than an explicit "use the default" value, it's not actually going to be an issue.  If it's meaning that we have to have more code in the library to try and catch those edge cases, I wouldn't worry much about it.

> Additionally, if you set max_credits here, you'd get a warning both from QuantumInstance.__init__ and RunConfig.__init__, even though the user is not directly creating RunConfig 

If the `stacklevel` of the warning is set correctly, they'll only see the warning that they triggered.  With the "default" warning filter in Python, you only see deprecation warnings that get stack-blamed to `__main__` - if the `stacklevel` causes the warning to point to library code, it's not shown with the default filters.